app/routes/circulars/circulars.server.ts

```diff
@@ -263,3 +273,280 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+/**
+ * Gets a specific version of a given circular
+ *
+ * Throws an HTTP error if:
+ *  - The provided circularId isNaN
+ *  - The provided version isNaN
+ *  - No version is found matching the provided circularId and version
+ *
+ * @param circularId
+ * @param version
+ * @returns a Circular corresponding to the specific version and Id
+ */
+export async function getCircularVersion(
+  circularId: number,
+  version: number
+): Promise<Circular> {
+  if (isNaN(circularId) || isNaN(version))
+    throw new Response(null, { status: 404 })
+  const db = await tables()
+  const result = await db.circulars_history.get({
+    circularId,
+    version,
+  })
+  if (!result)
+    throw new Response(null, {
+      status: 404,
+    })
+  return result
+}
+
+/**
+ * Adds a new entry to circulars_history for a given circular
+ * then updates the following fields of the Circular:
+ *  - body: Set to the provided body field
+ *  - subject: Set to the provided subject field
+ *  - editedBy: Set to the current user
+ *  - createdOn: Set to the current time
+ *  - version: Increment by 1
+ *
+ * Throws an HTTP error if:
+ *  - The current user is not a moderator
+ *  - The subject is invalid and not one of the valid options for
+ *    duplicates/mistaken submissions
+ *  - The body is invalid
+ *
+ * @param circular
+ * @param request
+ */
+export async function createNewCircularVersion(
+  circular: Omit<Circular, 'createdOn' | 'submitter'>,
+  request: Request
+) {
+  const user = await getUser(request)
+  if (!user?.groups.includes(moderatorGroup))
+    throw new Response('User is not in the submitters group', {
+      status: 403,
+    })
+
+  if (
+    !subjectIsValid(circular.subject) &&
+    !erroneousSubmissionKeywords.includes(circular.subject)
+  )
+    throw new Response('subject is invalid', { status: 400 })
```

Comment:
Not if the subject contains an `erroneousSubmissionKeyword`, but that `erroneousSubmissionKeywords` includes the subject.

If the subject is not valid and the subject is not literally one of the `erroneousSubmissionKeywords`: throw error

so if either the subject is valid, or the subject is one of the listed items in the `erroneousSubmissionKeywords`, it will continue, no?

Reply:
Oh, I get it now.